robot/impl/resource_manager.go

```diff
@@ -1067,6 +1125,9 @@ func (manager *resourceManager) markResourcesRemoved(
 			addNames(subG.Names()...)
 		}
 		manager.resources.MarkForRemoval(subG)
+		if err := manager.removeUniqueResource(rName.String()); err != nil {
```

Comment:
is there anywhere else you think we should do this? This made sense to me and seemed to work well.

Reply:
This should work for components, services and remotes, and I see that you have calls for processes and modules above. So, seems good to me.